rfc/0052-sdk3-full-text-search.md

```diff
@@ -606,18 +606,154 @@ interface SearchRowLocation {
 
 ### SearchFacetResult
 
-An individual facet result has both metadata and details, as each facet can define ranges into which results are categorized.
+The `facets` part of the result, a sibling to `hits`, contains information relative to the facets the user asked for. If no `facets` section was provided in the query, this section is omitted.
+
+Note that the whole `facets` section **is a single JSON object**. The `name` of each facet, which is defined at query time by the user, is represented in the attribute names.
+
+An individual facet result has both metadata and details, as each facet can define ranges into which results are categorized. The JSON representation of a single facet's results looks like this:
+
+```
+SearchFacetResult {
+    "field": string
+    "total": long
+    "missing": long
+    "other": long
+    "terms": [ {TermRange}, ... ] <1>
+    "numeric_ranges": [ {NumericRange}, ...] <1>
+    "date_ranges": [ {DateRange}, ... ] <1>
+}
+```
+(1) all mutually exclusive (`terms`, `numeric_ranges` and `date_ranges` cannot be combined).
+
+See the following JSON example:
+
+```
+"facets": {
+    "category": { <1>
+      "field": "style",
+          ...
+      "terms": [ ... ]
+    },
+    "strength": { <2>
+      "field": "abv",
+          ...
+      "numeric_ranges": [ ... ]
+    },
+    "updateRange": { <3>
+      "field": "updated",
+          ...
+      "date_ranges": [ ... ]
+    }
+  }
+```
+We omitted part of the section, but this shows that we have 3 facets: a _term_ facet **(1)** named "category", a _numeric_ facet **(2)** named "strength" and a _date_ facet **(3)** named "updateRange".
+
+The section could be represented by a `Map` (or equivalent, eg. associative array).
+
+```
+Map<String, SearchFacetResult> facets();
+```
+
+It could also be represented by any other adequate `Facets` class or structure, as long as the following operations are possible:
+
+ - iteration of all the facets
+ - retrieval of a facet by name
+
+In code, individual facet results could be represented by an interface and three concrete implementations, one for each facet type:
 
 ```
 interface SearchFacetResult {
-    String Name();
-    String Field();
-    uint64 Total();
-    uint64 Missing();
-    uint64 Other();
+    String name();
+    String field();
+    long total();
+    long missing();
+    long other();
+}
+```
+Note the `name` is made part of the `FacetResult`, so that it can be used in isolation.
+
+The results could also be implemented by a single concrete implementation:
+
+```
+class SearchFacetResult {
+    String name();  // deprecated
+    String field();
+    long total();
+    long missing();
+    long other();
+    List<TermFacetResult> terms();
```

Comment:
not in the pareent class, but in the child impl classes?

Reply:
Covered just above, this is for the single concrete implementation model.